goal_src/levels/citadel/citb-drop-plat-CIT.gc

```diff
@@ -5,3 +5,921 @@
+        )
+       (when (nonzero? s0-0)
+        (set! sv-64 (the int (* 150.0 (rand-vu))))
+        (set! sv-48 (get-process *default-dead-pool* drop-plat #x4000))
+        (set! (-> self child-array data s1-0) (ppointer->handle (when sv-48
+                                                                 (let
+                                                                  ((t9-6
+                                                                    (method-of-type
+                                                                     drop-plat
+                                                                     activate
+                                                                     )
+                                                                    )
+                                                                   )
+                                                                  (t9-6
+                                                                   (the-as
+                                                                    drop-plat
+                                                                    sv-48
+                                                                    )
+                                                                   self
+                                                                   'drop-plat
+                                                                   (the-as
+                                                                    pointer
+                                                                    #x70004000
+                                                                    )
+                                                                   )
+                                                                  )
+                                                                 (let
+                                                                  ((t9-7
+                                                                    run-function-in-process
+                                                                    )
+                                                                   (a0-8 sv-48)
+                                                                   (a1-5
+                                                                    drop-plat-init-by-other
+                                                                    )
+                                                                   (a2-4 gp-0)
+                                                                   )
+                                                                  (->
+                                                                   self
+                                                                   duration
+                                                                   )
+                                                                  ((the-as
```

Comment:
I think this cast is wrong, it should be `(function process function vector uint uint int)`.